addons/account/models/account_tax.py

```diff
@@ -225,6 +238,93 @@ def _compute_tax_group_id(self):
                 ('country_id', '=', False),
             ], limit=1)
 
+    def _hook_compute_is_used(self):
+        '''
+            To be overriden to add taxed transactions in the computation of `is_used`
+            Should return a Counter containing a dictionary {record: int} where
+            the record is an account.tax object. The int should be greater than 0
+            if the tax is used in a transaction.
+        '''
+        return Counter()
+
+    def _compute_is_used(self):
+        taxes_in_transactions_ctr = (
+            Counter(dict(self.env['account.move.line']._read_group([], groupby=['tax_ids'], aggregates=['__count']))) +
+            Counter(dict(self.env['account.reconcile.model.line']._read_group([], groupby=['tax_ids'], aggregates=['__count']))) +
+            self._hook_compute_is_used()
+        )
+        for tax in self:
+            tax.is_used = bool(taxes_in_transactions_ctr[tax])
+
+    @api.depends('repartition_line_ids.account_id', 'repartition_line_ids.factor_percent', 'repartition_line_ids.use_in_tax_closing', 'repartition_line_ids.tag_ids')
+    def _compute_repartition_lines_str(self):
+        for tax in self:
+            repartition_lines_str = tax.repartition_lines_str or ""
+            if tax.is_used:
+                for repartition_line in tax.repartition_line_ids:
+                    repartition_line_info = {
+                        _('id'): repartition_line.id,
+                        _('Factor Percent'): repartition_line.factor_percent,
+                        _('Account'): repartition_line.account_id.name or _('None'),
+                        _('Tax Grids'): repartition_line.tag_ids.mapped('name') or _('None'),
+                        _('Use in tax closing'): _('True') if repartition_line.use_in_tax_closing else _('False'),
+                    }
+                    repartition_lines_str += str(repartition_line_info) + '//'
+                repartition_lines_str = repartition_lines_str.strip('//')
+            tax.repartition_lines_str = repartition_lines_str
+
+    def _message_log_repartition_lines(self, old_value_str, new_value_str):
+        self.ensure_one()
+        if not self.is_used:
+            return
+
+        old_values = old_value_str.split('//')
+        new_values = new_value_str.split('//')
+
+        kwargs = {}
+        for old_value, new_value in zip(old_values, new_values):
+            if old_value != new_value:
+                old_value = ast.literal_eval(old_value)
+                new_value = ast.literal_eval(new_value)
+                diff_keys = [key for key in old_value if old_value[key] != new_value[key]]
+                repartition_line = self.env['account.tax.repartition.line'].search([('id', '=', new_value['id'])])
+                body = "<b>{type}</b> {rep} {seq}:<ul class='mb-0 ps-4'>{changes}</ul>".format(
```

Comment:
don't put a `Markup()` around the final result but create a Markup on the actual content, so
```suggestion
                body = Markup("<b>{type}</b> {rep} {seq}:<ul class='mb-0 ps-4'>{changes}</ul>").format(
```
(and you won't need these explicit `escape`

Reply:
@aboo-odoo if `changes` contains html, `Markup` should be put around the markup content. And don't forget that `''.join(Markup)` produces a string, not a Markup. See the doc for some examples:
https://www.odoo.com/documentation/master/developer/reference/backend/security.html#creating-safe-content-using-markup

Reply:
Suggestion have been applied :smile: Thanks again!